src/js/fs.js

```diff
@@ -264,16 +264,26 @@ fs.readFileSync = function(path) {
 };
 
 
-fs.writeFile = function(path, data, callback) {
+fs.writeFile = function(path, data, options, callback) {
+  if (util.isFunction(options)) {
+    callback = options;
+  }
   checkArgString(path);
   checkArgFunction(callback);
 
   var fd;
   var len;
   var bytesWritten;
   var buffer = ensureBuffer(data);
+  var flag = 'w';
```

Comment:
Maybe we could have the lines 278-284 and 328-334 move to a helper method so we can avoid code duplication.